src/System.ServiceModel.NetNamedPipe/src/System/ServiceModel/Channels/PipeConnectionInitiator.cs

```diff
@@ -0,0 +1,437 @@
+            NamedPipeClientStream namedPipeClient;
+            try
+            {
+                namedPipeClient = new NamedPipeClientStream(".", resolvedAddress, PipeDirection.InOut, PipeOptions.Asynchronous, TokenImpersonationLevel.Anonymous, HandleInheritability.None);
+                await namedPipeClient.ConnectAsync((int)backoffHelper.OriginalTimeout.TotalMilliseconds);
+            }
+            catch (Exception ex)
+            {
+                int error = PipeError.GetErrorFromHResult(ex.HResult);
+
+                if (error == UnsafeNativeMethods.ERROR_FILE_NOT_FOUND || error == UnsafeNativeMethods.ERROR_PIPE_BUSY)
+                {
+                    if (lastAttempt)
+                    {
+                        Exception innerException = new PipeException(SR.Format(SR.PipeConnectAddressFailed,
+                            resolvedAddress, ex.Message), ex.HResult);
+
+                        TimeoutException timeoutException;
+                        string endpoint = remoteUri.AbsoluteUri;
+
+                        if (error == UnsafeNativeMethods.ERROR_PIPE_BUSY)
+                        {
+                            timeoutException = new TimeoutException(SR.Format(SR.PipeConnectTimedOutServerTooBusy,
+                                endpoint, backoffHelper.OriginalTimeout), innerException);
+                        }
+                        else
+                        {
+                            timeoutException = new TimeoutException(SR.Format(SR.PipeConnectTimedOut,
+                                endpoint, backoffHelper.OriginalTimeout), innerException);
+                        }
+
+                        throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(timeoutException);
+                    }
+                }
+                else
+                {
+                    PipeException innerException = new PipeException(SR.Format(SR.PipeConnectAddressFailed,
+                        resolvedAddress, ex.Message), error);
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                        CreateConnectFailedException(remoteUri, innerException));
+                }
+
+                return null;
+            }
+
+            try
+            {
+                namedPipeClient.ReadMode = PipeTransmissionMode.Message;
+            }
+            catch(Exception ex)
+            {
+                PipeException innerException = new PipeException(SR.Format(SR.PipeModeChangeFailed, ex.Message), ex.HResult);
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(
+                    CreateConnectFailedException(remoteUri, innerException));
+
+            }
+
+            return new PipeConnection(namedPipeClient, _bufferSize);
+        }
+    }
+
+    internal unsafe class PipeSharedMemory : IDisposable
+    {
+        internal const string PipeLocalPrefix = @"Local\";
```

Comment:
I noticed that named pipes on my computer tend to capitalize `LOCAL\`. I'm guessing it doesn't matter.

Reply:
This was copied from WCF in NetFx so I think you're right. I suspect it's the whole case preserving case insensitive thing that Windows does for its file system by default.